vcd/resource_vcd_org_vdc_template_instance_test.go

```diff
@@ -0,0 +1,186 @@
+					resource.TestCheckResourceAttrPair(vdc, "description", instance, "description"),
+					resource.TestCheckResourceAttrPair(vdc, "org", "data.vcd_org.org", "name"),
+
+					// Correctness of what we put in the template (and default values)
+					resource.TestCheckResourceAttrPair(vdc, "allocation_model", template, "allocation_model"),
+					resource.TestCheckResourceAttrPair(vdc, "nic_quota", template, "nic_quota"),
+					resource.TestCheckResourceAttrPair(vdc, "vm_quota", template, "vm_quota"),
+					resource.TestCheckResourceAttrPair(vdc, "network_quota", template, "provisioned_network_quota"),
+					resource.TestCheckResourceAttrPair(vdc, "enable_thin_provisioning", template, "enable_thin_provisioning"),
+					resource.TestCheckResourceAttrPair(vdc, "enable_fast_provisioning", template, "enable_fast_provisioning"),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVdcTemplateInstanceStep1 = `
+data "vcd_org" "org" {
+  name = "{{.OrgToPublish}}"
+}
+
+data "vcd_provider_vdc" "pvdc" {
+  name = "{{.ProviderVdc}}"
+}
+
+data "vcd_external_network_v2" "ext_net" {
+  name = "{{.ExternalNetwork}}"
+}
+
+resource "vcd_org_vdc_template" "template" {
+  name               = "{{.Name}}"
+  tenant_name        = "{{.Name}}"
+  allocation_model   = "Flex"
+
+  compute_configuration {
+    cpu_allocated     = 0
+    cpu_limit         = 256
+    cpu_guaranteed    = 50
+    cpu_speed         = 1000
+    memory_allocated  = 1024
+    memory_limit      = 0
+    memory_guaranteed = 50
+    
+    elasticity                 = true
+    include_vm_memory_overhead = true
+  }
+
+  edge_gateway {
+    name = "{{.Name}}"
+    ip_allocation_count = 0
+    routed_network_name = "{{.Name}}"
+    routed_network_description = "{{.Name}}"
+    routed_network_gateway_cidr = "192.168.1.1/24"
+    static_ip_pool {
+      start_address = "192.168.1.10"
+      end_address = "192.168.1.20"
+    }
+  }
+
+  provider_vdc {
+    id                  = data.vcd_provider_vdc.pvdc.id
+    external_network_id = data.vcd_external_network_v2.ext_net.id
+  }
+
+  storage_profile {
+    name    = "{{.StorageProfile}}"
+    default = true
+    limit   = 0
+  }
+
+  enable_thin_provisioning = true
+  enable_fast_provisioning = true
+
+  readable_by_org_ids = [
+    data.vcd_org.org.id
+  ]
+}
+
+resource "vcd_org_vdc_template_instance" "instance" {
+  org_vdc_template_id = vcd_org_vdc_template.template.id
+  name                = vcd_org_vdc_template.template.name
+  org_id              = data.vcd_org.org.id
+  description         = vcd_org_vdc_template.template.name
+
+  delete_instantiated_vdc_on_removal = true
+  delete_force                       = true
+  delete_recursive                   = true
+}
+
+# This one depends on the VDC Template instance, so it waits for it to be finished creating the VDC
+data "vcd_org_vdc" "new_vdc" {
+	org  = data.vcd_org.org.name
+    name = vcd_org_vdc_template_instance.instance.name
```

Comment:
Text indentation is off

Reply:
Should be ok now, not sure what happened there with the tabs.